dexmaker/src/main/java/com/google/dexmaker/stock/ProxyBuilder.java

```diff
@@ -630,6 +631,43 @@ private static void check(boolean condition, String message) {
         for (MethodSetEntry entry : methodsToProxy) {
             results[i++] = entry.originalMethod;
         }
+
+        // Sort the results array so that they are returned by this method
+        // in a deterministic fashion.
+        Arrays.sort(results, new Comparator<Method>() {
+            @Override
+            public int compare(Method method1, Method method2) {
+                MethodId<? extends T, ?> methodId1 = getMethodId(method1);
```

Comment:
Is it possible to reuse something from the Method API?

One idea:
   return m1.toString().compareTo(m2.toString())

For further inspiration, here's a hidden function in Method.java:

      public static final Comparator<Method> ORDER_BY_SIGNATURE = new Comparator<Method>() {
          @Override public int compare(Method a, Method b) {
              if (a == b) {
                  return 0;
              }
              int comparison = a.getName().compareTo(b.getName());
              if (comparison == 0) {
                  comparison = a.artMethod.findOverriddenMethodIfProxy().compareParameters(
                          b.getParameterTypes());
                  if (comparison == 0) {
                      // This is necessary for methods that have covariant return types.
                      Class<?> aReturnType = a.getReturnType();
                      Class<?> bReturnType = b.getReturnType();
                      if (aReturnType == bReturnType) {
                          comparison = 0;
                      } else {
                          comparison = aReturnType.getName().compareTo(bReturnType.getName());
                      }
                  }
              }
              return comparison;
          }
      };